function_calling/src/tools.py

```diff
@@ -152,40 +152,95 @@ class ReplSchema(BaseModel):
 ## SQL tool
 # tool schema
 class QueryDBSchema(BaseModel):
-    "A database querying tool. Use this to generate sql querys and retrieve the results from a database. Input should be a natural language question to the db."
+    "A query generation tool. Use this to generate sql querys and retrieve the results from a database. Do not pass sql querys directly Input must be a natural language question or instruction."
 
-    query: str = Field(..., description='python code to execute to evaluate')
+    query: str = Field(..., description='natural language question or instruction.')
+
+
+def sql_finder(text):
+    # print(text)
+    sql_code_pattern = re.compile(r'```sql\s+(.*?)\s+```', re.DOTALL)
+    match = sql_code_pattern.search(text)
+    if match:
+        query = match.group(1)
+        return query
+    else:
+        code_pattern = re.compile(r'```\s+(.*?)\s+```', re.DOTALL)
+        match = code_pattern.search(text)
+        if match:
+            query = match.group(1)
+            return query
+        else:
+            raise Exception('No sql code found in LLM generation')
 
 
 @tool(args_schema=QueryDBSchema)
 def query_db(query):
-    """A database querying tool. Use this to generate sql querys and retrieve the results from a database. Input should be a natural language question to the db."""
+    """query generation tool. Use this to generate sql querys and retrieve the results from a database. Do not pass sql querys directly Input must be a natural language question or instruction."""
 
-    fine_tunned_sql_llm = 'Meta-Llama-3-70B-Instruct'  #'nsql-llama-2-7b'
-    llm = SambaverseEndpoint(
-        sambaverse_model_name='Numbers Station/nsql-llama-2-7b',
+    # print(query)
+    # Using Sambaverse expert as model for generating the SQL Query
+    llm = Sambaverse(
+        sambaverse_model_name='Meta/Meta-Llama-3-8B-Instruct',
         streaming=True,
         model_kwargs={
-            'max_tokens_to_generate': 200,
-            'select_expert': fine_tunned_sql_llm,
+            'max_tokens_to_generate': 512,
+            'select_expert': 'Meta-Llama-3-8B-Instruct',
             'temperature': 0.0,
             'repetition_penalty': 1.0,
             'top_k': 1,
             'top_p': 1.0,
             'do_sample': False,
+            'process_prompt': True,
         },
     )
 
+    # Using SambaStudio CoE expert as model for generating the SQL Query
+    # llm = SambaStudio(
+    #     streaming=True,
+    #     model_kwargs={
+    #         'max_tokens_to_generate': 512,
+    #         'select_expert': 'Meta-Llama-3-8B-Instruct',
+    #         'temperature': 0.0,
+    #         'repetition_penalty': 1.0,
+    #         'top_k': 1,
+    #         'top_p': 1.0,
+    #         'do_sample': False,
+    #     },
```

Comment:
Missing `process_prompt`?

Reply:
in SambaStudio the default is False, if true needs special input formatting so not added.
when langchain community v 0.2.6 gets released Sambaverse integration will be the same I'll update Sambaverse instantiation removing process prompt as well